requirements.txt

```diff
@@ -18,5 +18,5 @@ distributed<2022.10
 pytest>=3.6,<8.0
 pytest-runner
 pytest-cov
-xdsl<=0.9.2
+xdsl<=0.12
```

Comment:
I would suggest `~= 0.12`. Using `<=0.12` will not trigger pip to upgrade older versions.

Reply:
Or even just `==0.12` because everything breaks with every update anyway

Reply:
Using ~= for flexibility, as it's still a branch meant for iteration!